### PR TITLE
Bug 1853743: Support any image format for the profile picture on FedCM Select Account Dialog

### PR DESCRIPTION
**Pull request by titooan**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1853743

### CODE REVIEW

File: android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt [hunk 1]
```diff
@@ -412,19 +412,13 @@ fun String.trimmed(): String {
...
-
 /**
  * Returns a bitmap from its base64 representation.
  * Returns null if the string is not a valid base64 representation of a bitmap
- * @param prefix the prefix expected for the string to be considered a valid base64 bitmap string
  */
-private fun String.base64ToBitmap(prefix: String): Bitmap? {
-    if (!startsWith(prefix)) {
+fun String.base64ToBitmap(): Bitmap? {
+    val base64BitmapRegex = "(data:image/[^;]+;base64,)(.)*"
```
Comment (MatthewTighe) [inline android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:420]: 🚗 would love a comment to describe what this regex is doing

Comment (Amejia481) [reply android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:420]: nit: Maybe we should call `toRegex()` directly on the string as the variable name already contains **"*.Regex"**.

File: android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt [hunk 2]
```diff
@@ -412,19 +412,13 @@ fun String.trimmed(): String {
     return this.take(MAX_URI_LENGTH)
 }
 
-/**
- * Returns a bitmap from the base64 representation of a PNG.
- * Returns null if the string is not a valid base64 representation of a PNG
- */
-fun String.base64PngToBitmap(): Bitmap? = base64ToBitmap(PNG_URI_PREFIX)
```
Comment (Amejia481) [inline android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:419]: nit: I think we should also remove `PNG_URI_PREFIX` as it looks it's not used anymore?

Comment (Amejia481) [reply android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:419]: Sorry I found it's still relevant!

File: android-components/components/support/ktx/src/test/java/mozilla/components/support/ktx/kotlin/StringTest.kt [hunk 3]
```diff
@@ -525,13 +525,33 @@ class StringTest {
...
+        assertNotNull(validBase64JpegString.base64ToBitmap())
+        assertNotNull(validBase64JpgString.base64ToBitmap())
+        assertNotNull(validBase64AnythingString.base64ToBitmap())
+    }
+
+    @Test
+    fun `given invalid base64 image strings when converting them into bitmap then the result is null`() {
```
Comment (Amejia481) [inline android-components/components/support/ktx/src/test/java/mozilla/components/support/ktx/kotlin/StringTest.kt:548]: Normally, for tests we use `GIVEN`, `WHEN` and  `THEN` in uppercase :)

**Review by Amejia481 — APPROVED** [review]
Apart from the comments it LGTM!

File: android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt [hunk 4]
```diff
@@ -196,7 +195,12 @@ fun String.sha1(): String {
     return digest.joinToString(
         separator = "",
         transform = { byte ->
-            String(charArrayOf(characters[byte.toInt() shr 4 and 0x0f], characters[byte.toInt() and 0x0f]))
+            String(
+                charArrayOf(
+                    characters[byte.toInt() shr 4 and 0x0f],
+                    characters[byte.toInt() and 0x0f],
+                ),
+            )
```
Comment (Amejia481) [inline android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:203]: nit: Lets try to avoid adding unrelated changes :)

Comment (titooan) [reply android-components/components/support/ktx/src/main/java/mozilla/components/support/ktx/kotlin/String.kt:203]: I missed this autoformat from AndroidStudio, thanks for spotting it!
